src/pages/iou/request/step/IOURequestStepDistance.js

```diff
@@ -191,6 +216,9 @@ function IOURequestStepDistance({
             setShouldShowAtLeastTwoDifferentWaypointsError(true);
             return;
         }
+        if (!isCreatingNewRequest) {
+            transactionWasSaved.current = true;
+        }
```

Comment:
We should have checked if `isEditing` was `false` here as well. This caused https://github.com/Expensify/App/issues/52241